slack-lint-checks/src/main/java/slack/lint/JsonInflaterMoshiCompatibilityDetector.kt

```diff
@@ -139,9 +140,7 @@ class JsonInflaterMoshiCompatibilityDetector : Detector(), SourceCodeScanner {
 
     if (!isInstantiable(psiClass)) return false
 
-    return hasPublicNoArgConstructor(psiClass) ||
-      hasJsonClassGenerateAdapter(psiClass) ||
-      hasAdaptedBy(psiClass)
+    return hasPublicNoArgConstructor(psiClass) || psiClass.hasMoshiAnnotation()
```

Comment:
The public no-arg constructor will be reflectively serialized, which moshi doesn't try to support for kotlin classes in the way that gson might. Is this intentional? I might gate this with an option of some sort

Reply:
the public no-arg constructor check was intended for Java files in particular but I can go ahead and remove it if you don't think we need it